### DISCUSSION TITLE
Using this library in a next.js 15 app

### DISCUSSION THREAD

**Discussion by alibombaye** [govuk-react/govuk-react#1872] [Q&A]
Hello,

I'm just getting started using Next.js. Which is the framework of choice for the Server side rendering capabilities the project needs to meet GDS.

I've imported the govuk-react library and created a simple page with a simple `H1` header. However, I get the following error when running;

```
TypeError: createContext only works in Client Components. Add the "use client" directive at the top of the file to use it. Read more: https://nextjs.org/docs/messages/context-in-server-component
```

Now this looks due to styled-components. Is there a solution around this to use the components from this library as server-side components?

Thanks

**Comment by Aaron-Riley** [comment]
@alibombaye The error happens because govuk-react uses client-side features (like Context) that don’t work in Next.js Server Components. Adding "use client" at the top of your file fixes it by making it a Client Component. For best practice, I’d separate Server Components (for SSR/data fetching) and Client Components (for UI/interactivity) into different files. Mixing them can lead to performance hits or unexpected behavior.
